util/chunk/chunk_test.go

```diff
@@ -180,43 +180,48 @@ func (s *testChunkSuite) TestAppend(c *check.C) {
 	dst.Append(dst, 2, 6)
 	dst.Append(dst, 6, 6)
 
-	c.Assert(len(dst.columns), check.Equals, 3)
-
-	c.Assert(dst.columns[0].length, check.Equals, 12)
-	c.Assert(dst.columns[0].nullCount(), check.Equals, 6)
-	c.Assert(string(dst.columns[0].nullBitmap), check.Equals, string([]byte{0x55, 0x05}))
-	c.Assert(len(dst.columns[0].offsets), check.Equals, 0)
-	c.Assert(len(dst.columns[0].data), check.Equals, 4*12)
-	c.Assert(len(dst.columns[0].elemBuf), check.Equals, 4)
-
-	c.Assert(dst.columns[1].length, check.Equals, 12)
-	c.Assert(dst.columns[1].nullCount(), check.Equals, 6)
-	c.Assert(string(dst.columns[0].nullBitmap), check.Equals, string([]byte{0x55, 0x05}))
-	c.Assert(fmt.Sprintf("%v", dst.columns[1].offsets), check.Equals, fmt.Sprintf("%v", []int64{0, 3, 3, 6, 6, 9, 9, 12, 12, 15, 15, 18, 18}))
-	c.Assert(string(dst.columns[1].data), check.Equals, "abcabcabcabcabcabc")
-	c.Assert(len(dst.columns[1].elemBuf), check.Equals, 0)
-
-	c.Assert(dst.columns[2].length, check.Equals, 12)
-	c.Assert(dst.columns[2].nullCount(), check.Equals, 6)
-	c.Assert(string(dst.columns[0].nullBitmap), check.Equals, string([]byte{0x55, 0x05}))
-	c.Assert(len(dst.columns[2].offsets), check.Equals, 13)
-	c.Assert(len(dst.columns[2].data), check.Equals, 150)
-	c.Assert(len(dst.columns[2].elemBuf), check.Equals, 0)
+	assert.Equal(t, 3, dst.NumCols())
+	assert.Equal(t, 12, dst.NumRows())
+
+	col := dst.Column(0)
+	assert.Equal(t, 12, col.length)
+	assert.Equal(t, 6, col.nullCount())
+	assert.Equal(t, string([]byte{0b1010101, 0b0000101}), string(col.nullBitmap))
+	assert.Equal(t, 0, len(col.offsets))
+	assert.Equal(t, 4*12, len(col.data))
+	assert.Equal(t, 4, len(col.elemBuf))
+
+	col = dst.Column(1)
+	assert.Equal(t, 12, col.length)
+	assert.Equal(t, 6, col.nullCount())
+	assert.Equal(t, string([]byte{0b1010101, 0b0000101}), string(col.nullBitmap))
+	assert.Equal(t, fmt.Sprintf("%v", []int64{0, 3, 3, 6, 6, 9, 9, 12, 12, 15, 15, 18, 18}), fmt.Sprintf("%v", col.offsets))
+	assert.Equal(t, "abcabcabcabcabcabc", string(col.data))
+	assert.Equal(t, 0, len(col.elemBuf))
+
+	col = dst.Column(2)
+	assert.Equal(t, 12, col.length)
+	assert.Equal(t, 6, col.nullCount())
+	assert.Equal(t, string([]byte{0b1010101, 0b0000101}), string(col.nullBitmap))
+	assert.Equal(t, 13, len(col.offsets))
+	assert.Equal(t, (len(jsonObj.Value)+int(unsafe.Sizeof(jsonObj.TypeCode)))*6, len(col.data))
+	assert.Equal(t, 0, len(col.elemBuf))
 	for i := 0; i < 12; i += 2 {
-		jsonElem := dst.GetRow(i).GetJSON(2)
-		cmpRes := json.CompareBinary(jsonElem, jsonObj)
-		c.Assert(cmpRes, check.Equals, 0)
+		jsonElem := col.GetJSON(i)
```

Comment:
semantic changed?

Reply:
Yes. row then col -> col then row. Do I need to test this property?

Reply:
You can always give a follow up PR to make significant change and keep the migrate as small as possible. Try to avoid mix up thing which is hard to review.